components/camera/replaypcd/replaypcd.go

```diff
@@ -146,8 +148,28 @@ func (replay *pcdCamera) NextPointCloud(ctx context.Context) (pointcloud.PointCl
 		return nil, errEndOfDataset
 	}
 
+	// If the caller is communicating with the replay camera over gRPC, set the timestamps on
+	// the gRPC header.
+	md := resp.GetData()[0].GetMetadata()
+	if stream := grpc.ServerTransportStreamFromContext(ctx); stream != nil {
+		timeReq := md.GetTimeRequested()
```

Comment:
[nit] I prefer to place these above their associated if statements instead of at top, but up to you

Reply:
Works for me!